### PR TITLE
Add AWS v2 DynamoDB instrumentation

### PR DESCRIPTION
**Pull request by twcrone**
_Before contributing, please read our [contributing guidelines](https://github.com/newrelic/newrelic-java-agent/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/newrelic/.github/blob/main/CODE_OF_CONDUCT.md)._

### Overview
Add AWS v2 DynamoDB instrumentation to sync and async DynamoDB clients

### Related Github Issue
NA

### Testing
New tests added.  "tag" methods not able to be tested as LocalDynamoDB for tests does not implement tags.
[here](https://github.com/newrelic/newrelic-java-agent/blob/main/CONTRIBUTING.md),

### Checks

[X] Are your contributions backwards compatible with relevant frameworks and APIs?
[ ] Does your code contain any breaking changes? Please describe. 
[X] Does your code introduce any new dependencies? Please describe.

### CODE REVIEW

**Comment by twcrone** [comment]
Xi Xia and I are looking at the CF work on this right now.

**Comment by twcrone** [comment]
Ok, I think it was failing because we have "passesOnly" on our version for AWS v2 SDK or higher and it can apparently work with much earlier versions (even some 2.0..-preview versions) so made it expect the instrumentation verifier to pass with 2.1.0+ to go back a bit but not 2.0.

**Comment by twcrone** [comment]
Taking "draft" off PR.

**Review by XiXiaPdx — APPROVED** [review]
LGTM...but someone else probably should take a look at this. 

Also, lets run the AITs againts this branch before merging.

File: instrumentation/aws-java-sdk-dynamodb-2.15.34/src/main/java/software/amazon/awssdk/services/dynamodb/DefaultDynamoDbClient_Instrumentation.java
Comment (meiao) [inline instrumentation/aws-java-sdk-dynamodb-2.15.34/src/main/java/software/amazon/awssdk/services/dynamodb/DefaultDynamoDbClient_Instrumentation.java:123]: This var is not being used.
Guess this is leftover from a refactor that created the getEndpoint method.

File: instrumentation/aws-java-sdk-dynamodb-2.15.34/src/main/java/software/amazon/awssdk/services/dynamodb/DefaultDynamoDbAsyncClient_Instrumentation.java
Comment (meiao) [inline instrumentation/aws-java-sdk-dynamodb-2.15.34/src/main/java/software/amazon/awssdk/services/dynamodb/DefaultDynamoDbAsyncClient_Instrumentation.java:23]: is the query time being properly accounted for here?
Does Dynamo uses the same thread that called and returns a calculated future?

File: instrumentation/aws-java-sdk-dynamodb-2.15.34/src/main/java/software/amazon/awssdk/services/dynamodb/DefaultDynamoDbClient_Instrumentation.java
Comment (twcrone) [reply instrumentation/aws-java-sdk-dynamodb-2.15.34/src/main/java/software/amazon/awssdk/services/dynamodb/DefaultDynamoDbClient_Instrumentation.java:123]: ah yes...thanks!

File: instrumentation/aws-java-sdk-dynamodb-2.15.34/src/main/java/software/amazon/awssdk/services/dynamodb/DefaultDynamoDbAsyncClient_Instrumentation.java
Comment (twcrone) [reply instrumentation/aws-java-sdk-dynamodb-2.15.34/src/main/java/software/amazon/awssdk/services/dynamodb/DefaultDynamoDbAsyncClient_Instrumentation.java:23]: I honestly don't know.  @XiXiaPdx would need to help clear that up.

**Review by meiao — APPROVED** [review]

**Comment by twcrone** [comment]
Unit tests are failing now on "compileTestScala" after removing unused variable.

**Pull request merged**
